File: Data Structures & Algorithms/is-anagram/submission-3.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        count = {}

        for ch in s:
            count[ch] = count.get(ch, 0) + 1
            print(ch, count[ch])

        for ch in t:
            count[ch] = count.get(ch, 0) - 1
            print(ch, count[ch])

        for val in count.values():
            print(val)
            if val != 0:
                return False
        return True

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-3.cs
public class Solution {
    public int SubarraySum(int[] nums, int k) {
        var dict = new Dictionary<int, int>();
        dict[0] = 1;

        int currentSum = 0;
        int count = 0;

        foreach (int num in nums)
        {
            currentSum = currentSum + num;

            if (dict.ContainsKey(currentSum - k))
                count = count + dict[currentSum - k];
            
            if (dict.ContainsKey(currentSum))
                dict[currentSum]++;
            else
                dict[currentSum] = 1;
        }

        return count;
    }
}